src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,51 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { IconProps } from "@components/Icon/Icon";
+import { ReactElement } from "react";
+import css from "./Slider.module.css";
+
+export interface IconItem {
+    id: string;
+    icon: ReactElement<IconProps>;
+}
+
+export interface TextItem {
+    id: string;
+    name: string;
+}
+
+export interface SliderProps {
+    items: TextItem[] | IconItem[];
+    activeItemId: string;
+    onChange: (id: string) => void;
+}
+
+const isIconItem = (item: TextItem | IconItem): item is IconItem => (item as IconItem).icon !== undefined;
+const getItemClasses = (id: string, activeItemId: string) =>
+    [css.item, id === activeItemId ? css.active : ""].join(" ");
+
+export default function Slider({ items, activeItemId, onChange }: SliderProps): ReactElement<SliderProps> {
+    return (
+        <ul data-test-id="slider" className={css.wrapper}>
+            {items.map((item) =>
+                isIconItem(item) ? (
+                    <li
+                        key={item.id}
+                        onClick={() => onChange(item.id)}
```

Comment:
> The state change will be handled by the parent component so that the active item is also accessible from outside the Slider. I have updated the story to reflect a working example.

What are the pros of doing of moving out the active item id?
My way of doing it would be like:
```ts
const onItemClicked = (id: string) => {
	setActiveItemId(id) // Local useState
	onChange && onChange(id)
}

//[...]
<li
    key={item.id}
    onClick={() => onItemClicked(item.id)}
    className={getItemClasses(item.id, activeItemId)}
>
```